src/qt/paymentserver.cpp

```diff
@@ -645,10 +645,12 @@ void PaymentServer::fetchPaymentACK(CWallet* wallet, const SendCoinsRecipient& r
     else {
         CPubKey newKey;
         if (wallet->GetKeyFromPool(newKey)) {
-            CKeyID keyID = newKey.GetID();
-            wallet->SetAddressBook(keyID, strAccount, "refund");
+            // As we don't care about whether the other side understands a particular address type
+            // encoding, use the change type here.
```

Comment:
Maybe replace the comment with, if I understand correctly: "We'll use the change type here. In the BIP-70 payment protocol the destination for refunds is communicated as an output script, not an address. Therefore we don't need to care about whether the other side understands a particular address type."